modin/numpy/arr.py

```diff
@@ -137,9 +137,20 @@ def __init__(
         _query_compiler=None,
         _ndim=None,
     ):
+        self._siblings = []
         ErrorMessage.single_warning(
             "Using Modin's new NumPy API. To convert from a Modin object to a NumPy array, either turn off the ExperimentalNumPyAPI flag, or use `modin.utils.to_numpy`."
         )
+        if isinstance(object, array):
+            _query_compiler = object._query_compiler.copy()
```

Comment:
I followed the convention of the Dataframe class here, and copy the QC - but this creates a deep copy of the query compiler - should we really do this even when copy is False?

Reply:
Digging into the code, it seems that `QC.copy()` creates a new query compiler object that points to the exact same partitions, so I think its fine to keep the copy here?

Reply:
in that case, probably yeah, should be safe (and not too much memory, so better eat a little more mem than fail sometime down the drain)